.github/workflows/device-discovery-release.yaml

```
@@ -0,0 +1,245 @@
+name: Device Discovery Release
+on:
+  workflow_dispatch:
+  push:
+    branches:
+        - "release"
+    paths:
+        - "device-discovery/**"
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: false
+
+env:
+  GH_TOKEN: ${{ secrets.ORB_CI_GH_TOKEN }}
+  SEMANTIC_RELEASE_PACKAGE: ${{ github.repository }}
+  PYTHON_RUNTIME_VERSION: "3.11"
+  APP_NAME: device-discovery
+  PYTHON_PACKAGE_NAME: netboxlabs-device-discovery
+
+jobs:
+  get-python-package-name:
+    name: Get Python package name
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - name: Python package name
+        id: python-package-name
+        run: echo "python-package-name=${{ env.PYTHON_PACKAGE_NAME }}" >> "$GITHUB_OUTPUT"
+    outputs:
+        python-package-name: ${{ steps.python-package-name.outputs.python-package-name }}
+
+  get-next-version:
+    name: Semantic release get next version
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "lts/*"
+      - name: Write package.json
+        uses: DamianReeves/write-file-action@master
+        with:
+          path: ./package.json
+          write-mode: overwrite
+          contents: |
+            {
+              "name": "${{ env.APP_NAME }}",
+              "version": "1.0.0",
+              "devDependencies": {
+                "semantic-release-export-data": "^1.0.1",
+                "@semantic-release/changelog": "^6.0.3"
+              }
+            }
+      - name: Write .releaserc.json
+        uses: DamianReeves/write-file-action@master
+        with:
+          path: ./.releaserc.json
+          write-mode: overwrite
+          contents: |
+            {
+              "branches": "release",
+              "repositoryUrl": "https://github.com/netboxlabs/orb-discovery",
+              "debug": "true",
+              "tagFormat": "v${version}",
```

To produce tags per backend/app name, e.g. `device-discovery/v0.1.0`
```suggestion
              "tagFormat": "${{ env.APP_NAME }}/v${version}",
```